pkg/server/handler.go

```diff
@@ -155,11 +155,12 @@ func WithClusterScope(apiHandler http.Handler) http.HandlerFunc {
 func WithClusterAnnotation(handler http.Handler) http.HandlerFunc {
 	return http.HandlerFunc(func(w http.ResponseWriter, req *http.Request) {
 		cluster := request.ClusterFrom(req.Context())
+		ctx := kaudit.WithAuditAnnotations(req.Context())
```

Comment:
have this here is not good. Please make this explicit in server.go and add a comment why we need it.

Reply:
As a dedicated middleware or as an inline function (to grab hold of the `*request.Request`.

Reply:
Somewhere in https://github.com/kcp-dev/kcp/blob/main/pkg/server/server.go#L202, next to your `WithClusterAnnotation`.